feature/identity/src/test/java/com/revenuecat/purchases/identity/IdentityManagerTests.kt

```diff
@@ -148,6 +150,161 @@ class IdentityManagerTests {
         assertThat(expectedError).isNotNull
     }
 
+    @Test
+    fun `login fails with error if the appUserID is empty`() {
+        every {
+            mockBackend.logIn(stubAnonymousID, "", any(), captureLambda())
+        } just Runs
+        var receivedError: PurchasesError? = null
+        mockCachedAnonymousUser()
+        identityManager.logIn("", { _, _ -> }, { error -> receivedError = error })
+        assertThat(receivedError).isNotNull
+        assertThat(receivedError?.code).isEqualTo(PurchasesErrorCode.InvalidAppUserIdError)
+        verify(exactly = 0) {
+            mockBackend.logIn(any(), any(), any(), any())
+        }
+    }
+
+    @Test
+    fun `login fails with error if the appUserID is blank`() {
+        every {
+            mockBackend.logIn(stubAnonymousID, "   ", any(), captureLambda())
+        } just Runs
+        var receivedError: PurchasesError? = null
+        mockCachedAnonymousUser()
+        identityManager.logIn("   ", { _, _ -> }, { error -> receivedError = error })
+        assertThat(receivedError).isNotNull
+        assertThat(receivedError?.code).isEqualTo(PurchasesErrorCode.InvalidAppUserIdError)
+        verify(exactly = 0) {
+            mockBackend.logIn(any(), any(), any(), any())
+        }
+    }
+
+    @Test
+    fun `login passes backend errors`() {
+        every {
+            mockBackend.logIn(stubAnonymousID, "new", any(), captureLambda())
+        } answers {
+            lambda<(PurchasesError) -> Unit>().captured.invoke(
+                    PurchasesError(PurchasesErrorCode.InvalidCredentialsError)
+            )
+        }
+        var receivedError: PurchasesError? = null
+        mockCachedAnonymousUser()
+        identityManager.logIn("new", { _, _ -> }, { error -> receivedError = error })
+        assertThat(receivedError).isNotNull
+        assertThat(receivedError?.code).isEqualTo(PurchasesErrorCode.InvalidCredentialsError)
+    }
+
+    @Test
+    fun `login passes received created and purchaserInfo from backend`() {
+        val randomCreated: Boolean = Random.nextBoolean()
+        val mockPurchaserInfo: PurchaserInfo = mockk()
+        every {
+            mockBackend.logIn(stubAnonymousID, "new", captureLambda(), any())
+        } answers {
+            lambda<(PurchaserInfo, Boolean) -> Unit>().captured.invoke(
+                    mockPurchaserInfo, randomCreated
+            )
+        }
+        every { mockDeviceCache.cachePurchaserInfo(any(), any()) } just Runs
+        var receivedError: PurchasesError? = null
+        var receivedPurchaserInfo: PurchaserInfo? = null
+        var receivedCreated: Boolean? = null
+        mockCachedAnonymousUser()
+
+        identityManager.logIn("new", {
+            purchaserInfo, created ->
```

Comment:
I would put this line in the previous one or just put all arguments in separate lines